protocol/common.go

```diff
@@ -1,3 +1,7 @@
+// Package protocol has no obvious theme and doesn't have much to do with a protocol. The
+// EthereumAssetAmount seems like it belongs with the coins types in the common package.
+// KeysAndProof seems like it belongs in some crypto package. ConvertContractSwapToMsg
+// seems like it belongs in the net/message package.
```

Comment:
I tried to figure out what the theme was, but my thought was that it might be better to just reorganize the code. Let me know what you think. I could be wrong.

Reply:
Package protocol has functions that are used by both the maker and taker during execution of the swap.